Intermittently failing MembershipUpdateTest.connectionsToTerminatedMember_shouldBeClosed

The `MembershipUpdateTest.connectionsToTerminatedMember_shouldBeClosed` failed on PR builder (`master` branch):
http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/1937

**Stack trace:**
```
java.lang.AssertionError: expected null, but was:<MockConnection{localEndpoint=[127.0.0.1]:5703, remoteEndpoint=[127.0.0.1]:5702, alive=false}>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotNull(Assert.java:755)
	at org.junit.Assert.assertNull(Assert.java:737)
	at org.junit.Assert.assertNull(Assert.java:747)
	at com.hazelcast.internal.cluster.impl.MembershipUpdateTest$18.run(MembershipUpdateTest.java:849)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1303)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1405)
	at com.hazelcast.internal.cluster.impl.MembershipUpdateTest.connectionsToTerminatedMember_shouldBeClosed(MembershipUpdateTest.java:846)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockNetworkingService.java
+++ b/hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockNetworkingService.java
@@ -157,8 +157,21 @@ class MockNetworkingService
 
             remoteConnection.localConnection = thisConnection;
             thisConnection.localConnection = remoteConnection;
+
+            if (!remoteConnection.isAlive()) {
+                // targetNode is not alive anymore.
+                suspectAddress(remote);
+                return null;
+            }
+
             ns.mapConnections.put(remote, remoteConnection);
             ns.logger.info("Created connection to endpoint: " + remote + ", connection: " + remoteConnection);
+
+            if (!remoteConnection.isAlive()) {
+                // If connection is not alive after inserting it into connection map,
+                // that means remote node is being stopping during connection creation.
+                suspectAddress(remote);
+            }
             return remoteConnection;
         }
 

